spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +66,26 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    @GetMapping( "/api/gateways/{gatewayId}/receiveApiKeyList")
+    public ResponseEntity<GatewayUpdate> findServiceByGatewayId(@PathVariable("gatewayId") String gatewayId) {
+        Proxy proxy = proxyRepository.findById(gatewayId);
+        GatewayUpdate gatewayUpdate;
+        Timestamp now = new Timestamp(System.currentTimeMillis());
+
+        if(proxy != null){
+            ServiceApi[] serviceApis = proxy.getServiceApis().stream().toArray(ServiceApi[]::new);
+            ApiKey[] apiKeys = proxy.getServiceApis().stream().flatMap(x -> x.getApiKeys()
```

Comment:
The `x` is quite confusing/generic. Either use `service` or at least `s` to improve readability. 

An even better option would be `ServiceApi::getApiKeys`.